src/preprocessing/passes/ackermann.cpp

```diff
@@ -187,9 +188,110 @@ void collectFunctionsAndLemmas(FunctionToArgsMap& fun_to_args,
 
 /* -------------------------------------------------------------------------- */
 
+/* Given a minimum capacity for an uninterpreted sort, return the size of the
+ * new BV type */
+size_t getBVSkolemSize(size_t capacity) { return (size_t)log2(capacity) + 1; }
+
+/* Given the lowest capacity requirements for each uninterpreted sorts, assign
+ * a sufficient bit-vector size. Get the converting map */
+void collectUSortsToBV(const vector<TNode>& terms,
+                       USortToBVSizeMap& usortCardinality,
+                       SubstitutionMap& sortsToSkolem)
+{
+  NodeManager* nm = NodeManager::currentNM();
+
+  for (TNode term : terms)
+  {
+    TypeNode type = term.getType();
+    size_t size = getBVSkolemSize(usortCardinality[type]);
+    Node skolem = nm->mkSkolem(
+        "BVSKOLEM$$",
+        nm->mkBitVectorType(size),
+        "a variable created by the ackermannization "
+        "preprocessing pass, representing a term with uninterpreted sort.");
+    sortsToSkolem.addSubstitution(term, skolem);
+  }
+}
+
+/* This function returns the list of terms with uninterpreted sort in formula
+ * represented by assertions. We use a BFS to get all terms without
+ * duplications. */
```

Comment:
Remove comment about BFS, since we don't care about the implementation of `expr::getVariables()` here.